test/distributed/_sharding_spec/test_sharding_spec.py

```diff
@@ -0,0 +1,54 @@
+import torch
+from torch.testing._internal.common_utils import TestCase
+from torch.distributed._sharding_spec import (
+    ChunkShardingSpec,
+    DevicePlacement
+)
+
+class TestShardingSpec(TestCase):
+
+    def test_device_placement(self):
```

Comment:
Since you have already reused `_parse_remote_device` util function, it will be great to also share the duplicate test cases for validation (probably by sharing two const lists to represent valid and invalid inputs). We may even consider reusing `DevicePlacement` in `RemoteModule` class, as `DevicePlacement` is more generic than tensor sharding.